kyaml/openapi/kubernetesapi/swagger.json

```diff
@@ -5098,6 +5098,7 @@
           },
           "type": "array",
           "x-kubernetes-list-map-keys": [
+            "name",
```

Comment:
I don't think that's correct. The name is missing in too many occasions because of decisions made initially, so it can't be used reliably as a key.

Reply:
I'm a little confused by your comment, could you clarify?

The goal here is to merge on name if it is there, and fall back on containerPort if the name is missing. That's why we are switching to x-kubernetes-list-map-keys, so that we can iterate through the list.

Reply:
I think you should merge on port/protocol, not name. I don't think that'd work the way you want.

Reply:
How would you propose to resolve #2767 if not merge on name?